SC_redesign/Crop_and_Soil/crop/leaf_area_index.py

```diff
@@ -31,30 +32,80 @@ def __init__(self):
 
     def grow_canopy(self):
         """main leaf area index function"""
-        self.determine_optimal_leaf_area_fraction()
+        self._lai_shapes = self.determine_lai_shapes(self.first_heat_fraction_point, self.second_heat_fraction_point,
+                                                     self.first_leaf_fraction_point, self.second_leaf_fraction_point)
+
+        self.optimal_leaf_area_fraction = self.determine_optimal_leaf_area_fraction(self.heat_fraction,
+                                                                                    self._lai_shapes[0],
+                                                                                    self._lai_shapes[1])
+
         self.determine_canopy_height()
-        self.check_for_senescence()
+        self.is_in_senescence = self.check_for_senescence
```

Comment:
I believe that `@property` functions don't need to be assigned to variables.
```suggestion
        self.is_in_senescence
```

Reply:
I actually think I was wrong - after looking at the code in its entirety, I think this line can just be deleted. The next line (45) contains the code `if self.is_in_senescence:`, which will automatically check via the `@property` (I think).